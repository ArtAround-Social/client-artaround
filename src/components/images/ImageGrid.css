.div {
  float: right;
  display: flex;
  flex-wrap: wrap;
  padding: 5px;
  width: 600px;
  background-color: rgb(252, 252, 252);
  max-height: 800px;
  overflow: auto;
  border-radius: 5px;
}

img {
  margin: 2px;
  max-width: 580px;
}

.images {
  flex: 50%;
  padding: 0 4px;
}

h3 {
  margin-bottom: 2px
}

::-webkit-scrollbar {
  width: 10px;
}

::-webkit-scrollbar-track {
  background: #f1f1f1; 
}
 
::-webkit-scrollbar-thumb {
  background: rgba(136, 136, 136, 0.788); 
}

::-webkit-scrollbar-thumb:hover {
  background: rgba(85, 85, 85, 0.76); 
}