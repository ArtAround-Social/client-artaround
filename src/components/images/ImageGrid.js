import React from 'react';
import PropTypes from 'prop-types';
import styles from './ImageGrid.css';

function ImageGrid({ images }) {
  const imageElement = images.map(image => (
    <img key={image.id} src={image.url}/>
  ));
  
  return(
    <div className={styles.div}>{imageElement}</div>
  );
}

ImageGrid.propTypes = {
  images: PropTypes.arrayOf(PropTypes.shape({
    image: PropTypes.string.isRequired
  }))
};

export default ImageGrid;
