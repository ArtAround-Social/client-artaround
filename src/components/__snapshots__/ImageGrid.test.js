import React from 'react';
import { shallow } from 'enzyme';
import ImageGrid from '../images/ImageGrid';

describe('ImageGrid component', () => {
  it('renders images', () => {
    const wrapper = shallow(
      <ImageGrid
        images = { [
          { id:1, url:'https://unsplash.com/photos/yC-Yzbqy7PY' },
          { id:2, url:'https://unsplash.com/photos/LNRyGwIJr5c' },
          { id:3, url:'https://unsplash.com/photos/6J--NXulQCs' },
          { id:4, url:'https://unsplash.com/photos/pwaaqfoMibI' },
        ] }
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
