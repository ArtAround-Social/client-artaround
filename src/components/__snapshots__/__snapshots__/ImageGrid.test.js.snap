// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImageGrid component renders images 1`] = `
<div
  className="div"
>
  <img
    key="1"
    src="https://unsplash.com/photos/yC-Yzbqy7PY"
  />
  <img
    key="2"
    src="https://unsplash.com/photos/LNRyGwIJr5c"
  />
  <img
    key="3"
    src="https://unsplash.com/photos/6J--NXulQCs"
  />
  <img
    key="4"
    src="https://unsplash.com/photos/pwaaqfoMibI"
  />
</div>
`;
